corehq/apps/hqwebapp/two_factor_gateways.py

```diff
@@ -116,23 +138,19 @@ def get_ip_address():
     method = device.method if isinstance(device, PhoneDevice) else None
 
     if ip_address and username and number and method:
-        if two_factor_rate_limiter_per_ip.allow_usage('ip:{}'.format(ip_address)) \
-                and two_factor_rate_limiter_per_user.allow_usage('user:{}'.format(username)) \
-                and two_factor_rate_limiter_per_number.allow_usage('number:{}'.format(number)) \
-                and global_two_factor_setup_rate_limiter.allow_usage():
-            two_factor_rate_limiter_per_ip.report_usage('ip:{}'.format(ip_address))
-            two_factor_rate_limiter_per_user.report_usage('user:{}'.format(username))
-            two_factor_rate_limiter_per_number.report_usage('number:{}'.format(number))
-            global_two_factor_setup_rate_limiter.report_usage()
-            status = _status_accepted
-        else:
-            status = _status_rate_limited
+
+        status, window = _check_for_exceeded_rate_limits(ip_address, number, username)
+        if status == _status_accepted:
+            _report_usage(ip_address, number, username)
+
     else:
+        window = None
         status = _status_bad_request
 
     metrics_counter('commcare.two_factor.setup_requests', 1, tags={
         'status': status,
         'method': method,
+        'window': window,
```

Comment:
is it ok to pass None into this?

Reply:
I'd pick a value like 'none' and pass it in directly i.e. `window or 'none'`. I think it'll cast it to string (`'None'`) and then lower case it (`'none'`) anyway, but might as well be explicit about it.